crates/wasmparser/src/validator.rs

```diff
@@ -1792,7 +1791,19 @@ mod arc {
                 return None;
             }
             debug_assert!(Arc::get_mut(&mut self.arc).is_some());
-            Some(unsafe { &mut *(&*self.arc as *const T as *mut T) })
+            Some(
+                // # Safety
+                //
+                // This operation is safe since the API asserts that
+                // `self.owned` is only ever `true` if there is just
+                // one owner of the `Arc`.
+                // As soon as `Arc::get_mut_unchecked` we should
+                // make use of it instead.
```

Comment:
This `# Safety` comment doesn't really add anything over the structure of the code that's already here, so I don't think it should be added.